platforms/common/src/CommonKernels.cpp

```diff
@@ -2083,16 +2086,35 @@ void CommonCalcCustomNonbondedForceKernel::initialize(const System& system, cons
     }
 }
 
+// the shifting method previously used in initInteractionGroups
+// is undefined for signed types, hence we switched
+// to unsigned types for repeatability, and a special case switch operator
+// see https://en.cppreference.com/w/c/language/operator_arithmetic
+template<typename tileflags>
+tileflags inline fullShift(tileflags operand, int operant)
+{
+    if (operant == sizeof(tileflags) * 8) {
+        // shifting by the size of the operand is undefined
+        // the intent here appears to be to return a -0
+        return tileflags(0);
+    }
+    return (operand<<operant);
+}
+
+
+template <typename tileflags, typename tileflags4>
 void CommonCalcCustomNonbondedForceKernel::initInteractionGroups(const CustomNonbondedForce& force, const string& interactionSource, const vector<string>& tableTypes) {
     // Process groups to form tiles.
-    
+
+    const int tileSize = cc.getTileSize();
```

Comment:
The changes in this class are based on the assumption that the tile size used for computing interaction groups is the same as the size used for standard nonbonded interactions.  They both happen to be 32, but there's no reason they need to be the same.  They're computed with different code using different neighbor lists.  I think it's best to leave it as it is, with the size just hardcoded to 32.  Some day we might (or might not) decide it would be useful to make that configurable, but either way it shouldn't be required to match the main nonbonded kernel.